dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/NativeHttpRestClient.java

```diff
@@ -0,0 +1,386 @@
+        List<NamedXContentRegistry.Entry> entries = map.entrySet().stream()
+                .map(entry -> new NamedXContentRegistry.Entry(Aggregation.class, new ParseField(entry.getKey()), entry.getValue()))
+                .collect(Collectors.toList());
+        entries.add(new NamedXContentRegistry.Entry(
+                Suggest.Suggestion.class, new ParseField(TermSuggestionBuilder.SUGGESTION_NAME),
+                (parser, context) -> TermSuggestion.fromXContent(parser, (String)context)));
+        entries.add(new NamedXContentRegistry.Entry(Suggest.Suggestion.class, new ParseField(
+                PhraseSuggestionBuilder.SUGGESTION_NAME),
+                (parser, context) -> PhraseSuggestion.fromXContent(parser, (String)context)));
+        entries.add(new NamedXContentRegistry.Entry(Suggest.Suggestion.class, new ParseField(
+                CompletionSuggestionBuilder.SUGGESTION_NAME),
+                (parser, context) -> CompletionSuggestion.fromXContent(parser, (String)context)));
+        return entries;
+    }
+
+    public NativeHttpRestClient(final SSLContext sslContext,
+            final CredentialsProvider credentialsProvider, final String hostName, final int port) {
+
+        final String protocol = sslContext != null ? "https://" : "http://";
+
+        PoolingHttpClientConnectionManager cm;
+        SSLConnectionSocketFactory socketFactory = null;
+
+        if (sslContext != null) {
+            socketFactory = new SSLConnectionSocketFactory(sslContext);
+
+            Registry<ConnectionSocketFactory> socketFactoryRegistry = RegistryBuilder.<ConnectionSocketFactory>create()
+                    .register("https", socketFactory).build();
+
+            cm = new PoolingHttpClientConnectionManager(socketFactoryRegistry);
+
+        } else {
+            cm = new PoolingHttpClientConnectionManager();
+        }
+        // Increase max total connection to 200
+        cm.setMaxTotal(200);
+        // Increase default max connection per route to 20
+        cm.setDefaultMaxPerRoute(20);
+        // Increase max connections for localhost:80 to 50
+        HttpHost localhost = new HttpHost(hostName, port);
+        cm.setMaxPerRoute(new HttpRoute(localhost), 50);
+
+        if (socketFactory != null) {
+
+            client = HttpClients.custom().setDefaultCredentialsProvider(credentialsProvider)
+                    .setRetryHandler(new MyRequestRetryHandler())
+                    .setSSLSocketFactory(socketFactory)
+                    .setConnectionManager(cm)
+                    .build();
+        } else {
+            client = HttpClients.custom().setDefaultCredentialsProvider(credentialsProvider)
+                    .setRetryHandler(new MyRequestRetryHandler())
+                    .setConnectionManager(cm)
+                    .build();
+        }
+        NativeHttpRestClient.url = protocol + hostName + ":" + port;
+
+        List<NamedXContentRegistry.Entry> namedXContentEntries = Collections.emptyList();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'namedXContentEntries' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5121287)